airbyte-integrations/connectors/source-github/source_github/streams.py

```diff
@@ -0,0 +1,405 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+
+import tempfile
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional
+
+import requests
+import vcr
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.streams.http import HttpStream
+from requests.exceptions import HTTPError
+
+cache_file = tempfile.NamedTemporaryFile()
+
+
+class GithubStream(HttpStream, ABC):
+    url_base = "https://api.github.com/"
+
+    primary_key = "id"
+
+    # GitHub pagination could be from 1 to 100.
+    page_size = 100
+
+    # These fields will be used for data clearing. Put here keys which represent
+    # objects `{}`, like `user`, `actor` etc.
+    object_fields = ()
+
+    # These fields will be used for data clearing. Put here keys which represent
+    # lists `[]`, like `labels`, `assignees` etc.
+    list_fields = ()
+
+    def __init__(self, repository: str, **kwargs):
+        super().__init__(**kwargs)
+        self.repository = repository
+        self._page = 1
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        if response.json():
```

Comment:
I'm pretty sure we'll make one extra request in this case.

Reply:
Why?

Reply:
You check that we will not receive any data.
If we receive 77 records, then you will make another request for data.
Why not check that the data that will come to us will be less than `self._page_size`?

Reply:
Got it, you are right.

Reply:
Done.